testdata/p4_16_samples/issue363-bmv2.p4

```diff
@@ -0,0 +1,88 @@
+#include <v1model.p4>
+
+struct alt_t {
+  bit<1> valid;
+  bit<7> port;
+};
+
+struct row_t {
+  alt_t alt0;
+  alt_t alt1;
+};
+
+struct local_metadata_t {
+  row_t row;
+};
+
+header bitvec_hdr {
+  row_t row;
+}
+
+struct parsed_packet_t {
+    bitvec_hdr bvh;
+    bitvec_hdr bvh1;    
+};
+
+struct tst_t {
+    bitvec_hdr bvh0;
+}
+
+parser parse(packet_in pk, out parsed_packet_t h,
+             inout local_metadata_t local_metadata,
+             inout standard_metadata_t standard_metadata) {
+  state start {
+//      pk.extract(h.bvh);
+//    pk.extract(h.bvh1);
+    transition accept;
+  }
+}
+
+control ingress(inout parsed_packet_t h,
+                inout local_metadata_t local_metadata,
+	        inout standard_metadata_t standard_metadata) {
+    apply {
+        tst_t s;
+
+        // bit<48> mac;
+        // h.bvh.mac.macDA.lower28Bits = 0;
+        // h.bvh.mac.macDA.upper20Bits = 0;
+
+        local_metadata.row.alt0 = local_metadata.row.alt1;
+        local_metadata.row.alt0.valid = 1;
+        local_metadata.row.alt1.port = local_metadata.row.alt1.port + 1;
+        clone3(CloneType.I2E, 0, local_metadata.row);
+
+        // mac = h.bvh.mac.macDA.upper20Bits ++ h.bvh.mac.macDA.lower28Bits;
+/*
+        Cast support is TODO for bmv2.
+        sip.macDA = (macDA_t) mac;
+        mac       = (bit<48>) sip.macDA;
+*/
+    }
+}
+
+control egress(inout parsed_packet_t hdr,
+               inout local_metadata_t local_metadata,
+	       inout standard_metadata_t standard_metadata) {
+  apply { }
+}
+
+control deparser(packet_out b, in parsed_packet_t h) {
+  apply {
+//     b.emit(h.bvh);
```

Comment:
you should test the emit code.

Reply:
Sure.  It was commented out to debug issues.  With the next commit, all code in this P4 program will be tested.

Reply:
Yes.  It was commented out temporarily waiting for next round of code changes to test it.